docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
 # Developer Guide
 
+## Table of Content
+- [Acknowledgements](#acknowledgements)
+- [Introduction](#introduction)
+- [Architecture](#architecture)
+- [Design & Implementation](#design--implementation)
+  - [Study Session Implementation](#study-session-implementation)
+  - [Timetable Implementation](#timetable-implementation)
+  - [Storage Implementation](#storage-implementation)
+- [Product Scope](#product-scope)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- Team member Jun Lim's individual project codebase - [Github](#https://github.com/jltha/ip)
+
+## Introduction
+
+Our product, Sherpass, is a desktop application for students to manage their academic schedules by adding their tasks,
+Sherpass will then compile their tasks and present it to them in a timetable. Sherpass also provides a 
+platform for them to focus on their academic tasks through the use of timers in our study sessions. Students can use
+the time to block out time to focus on their tasks. Optimised for use via a Command Line Interface (CLI),
+users who can type fast will be able to plan out their tasks in a much quicker fashion compared to traditional GUI apps.
+
+## Architecture
+
+![image](https://user-images.githubusercontent.com/69501969/160375887-d6da7278-5259-4458-83c7-f53d89fef640.png)
+
+The above diagram provides a general overview of Sherpass and its major components. The four key areas are:
+
+- User Interface (UI)
+- Storage
+- Timetable
+- Study session
+
+### UI
+
+UI component consists the `Ui` class which manages interaction (receiving inputs and showing outputs) between the user 
+and the application.
+
+### Storage
+
+Storage component consists `Storage` and `StorageParser` classes.
+`Storage` class handles loading, writing and saving
+data to and from a JSON file, such that users' data will be saved automatically.
+`StorageParser` class handles the
+parsing of JSON from the saved data file.
+
+### Timetable
+
+For components with more complicated use-cases (`Task` and `Timer`), we separate an extra Logic class to achieve better
+modularity, such that each class addresses a separate concern.
+
+Timetable component consists of `Timetable`, `Task`, `TaskList`, `TaskParser`, `TaskLogic` and various commands.
+
+### Study session
+
+Timetable component consists of `Timer`, `Stopwatch`, `Countdown`, `TimerParser`, `TimerLogic` and various commands.
 
 ## Design & implementation
```

Comment:
i suggest splitting design and implementation, where design can include stuff like architecture overview,class diagrams of the 4 major compnents above